Client/Frontend/UserContent/UserScripts/Scripts_Dynamic/ScriptHandlers/Sandboxed/SiteStateListenerScriptHandler.swift

```diff
@@ -61,61 +61,67 @@ class SiteStateListenerScriptHandler: TabContentScript {
-        }
-        
-        let styleSelectorObjects = styleSelectors.map { selector, rules -> UserScriptType.SelectorsPollerSetup.StyleSelectorEntry in
-          UserScriptType.SelectorsPollerSetup.StyleSelectorEntry(
-            selector: selector, rules: rules
+      if let pageData = tab.currentPageData {
+        Task { @MainActor in
+          let domain = pageData.domain(persistent: !tab.isPrivate)
+          let models = await AdBlockStats.shared.cosmeticFilterModels(forFrameURL: frameURL, domain: domain)
+          let args = try await self.makeArgs(from: models, frameURL: frameURL)
+          let source = try ScriptFactory.shared.makeScriptSource(of: .selectorsPoller).replacingOccurrences(of: "$<args>", with: args)
+          
+          let secureSource = CosmeticFiltersScriptHandler.secureScript(
+            handlerName: CosmeticFiltersScriptHandler.messageHandlerName,
+            securityToken: CosmeticFiltersScriptHandler.scriptId,
+            script: source
+          )
+          
+          webView.evaluateSafeJavaScript(
+            functionName: secureSource,
+            frame: message.frameInfo,
+            contentWorld: .defaultClient,
+            asFunction: false,
+            completion: { _, error in
+              guard let error = error else { return }
+              Logger.module.error("\(error.localizedDescription)")
+            }
           )
         }
-        
-        let setup = UserScriptType.SelectorsPollerSetup(
-          frameURL: frameURL,
-          genericHide: models.contains { $0.genericHide },
-          hideSelectors: hideSelectors,
-          styleSelectors: Set(styleSelectorObjects)
-        )
-        
-        let encoder = JSONEncoder()
-        let data = try encoder.encode(setup)
-        let args = String(data: data, encoding: .utf8)!
-        let source = try ScriptFactory.shared.makeScriptSource(of: .selectorsPoller).replacingOccurrences(of: "$<args>", with: args)
-        let secureSource = CosmeticFiltersScriptHandler.secureScript(
-          handlerName: CosmeticFiltersScriptHandler.messageHandlerName,
-          securityToken: CosmeticFiltersScriptHandler.scriptId,
-          script: source
-        )
-        
-        webView.evaluateSafeJavaScript(
-          functionName: secureSource,
-          frame: frameInfo,
-          contentWorld: .defaultClient,
-          asFunction: false,
-          completion: { _, error in
-            guard let error = error else { return }
-            Logger.module.error("\(error.localizedDescription)")
-          }
-        )
-        
-        tab.currentPageData?.framesInfo.removeValue(forKey: frameURL)
       }
     } catch {
       assertionFailure("Invalid type of message. Fix the `Site.js` script")
       Logger.module.error("\(error.localizedDescription)")
     }
   }
+  
+  private func makeArgs(from models: [CosmeticFilterModel], frameURL: URL) async throws -> String {
+    return try await Task.detached(priority: .high) {
```

Comment:
You don't need this Task.detached since async methods without specific actor isolation will run on the global executor by default

Reply:
Ahh that is really useful information. I'll make the change. What about the priority?

Reply:
Priority is inherited from the parent `Task`, so if the parent `Task` is created with `high` priority then this will inherit that :) 

Verified with this bit of code: 

```swift
func foo() async throws {
  await Task(priority: .high) { @MainActor in
    print("Task", "current priorty", Task.currentPriority, Thread.isMainThread)
    await bar()
  }.value
}

func bar() async {
  print("test", "current priorty", Task.currentPriority, Thread.isMainThread)
}
```

prints:
```
Task current priorty TaskPriority(rawValue: 25) true
test current priorty TaskPriority(rawValue: 25) false
```

Reply:
@cuba Sorry if I wasn't clear, you don't need a child `Task` at all. Not just that you don't need `detached`